src/calliope/io.py

```diff
@@ -208,5 +216,116 @@ def save_csv(
+
+
+def read_rich_yaml(yaml: str | Path, allow_override: bool = False) -> AttrDict:
+    """Returns an AttrDict initialised from the given YAML file or string.
+
+    Uses calliope's "flavour" for YAML files.
+
+    Args:
+        yaml (str | Path): YAML file path or string.
+        allow_override (bool, optional): Allow overrides for already defined keys. Defaults to False.
+
+    Raises:
+        ValueError: Import solving requested for non-file input YAML.
+    """
+    if isinstance(yaml, str) and not os.path.exists(yaml):
+        yaml_path = None
+        yaml_text = yaml
+    else:
+        yaml_path = Path(yaml)
+        yaml_text = yaml_path.read_text(encoding="utf-8")
+
+    yaml_dict = AttrDict(_yaml_load(yaml_text))
+    yaml_dict = _resolve_yaml_imports(
+        yaml_dict, base_path=yaml_path, allow_override=allow_override
+    )
+    return yaml_dict
+
+
+def _yaml_load(src):
+    """Load YAML from a file object or path with useful parser errors."""
+    yaml = ruamel_yaml.YAML(typ="safe")
+    if not isinstance(src, str):
+        try:
+            src_name = src.name
+        except AttributeError:
+            src_name = "<yaml stringio>"
+        # Force-load file streams as that allows the parser to print
+        # much more context when it encounters an error
+        src = src.read()
+    else:
+        src_name = "<yaml string>"
+    try:
+        result = yaml.load(src)
+        if not isinstance(result, dict):
+            raise ValueError(f"Could not parse {src_name} as YAML")
+        return result
+    except ruamel_yaml.YAMLError:
+        logger.error(f"Parser error when reading YAML from {src_name}.")
+        raise
+
+
+def _resolve_yaml_imports(
+    loaded: AttrDict, base_path: str | Path | None, allow_override: bool
+) -> AttrDict:
+    loaded_dict = loaded
+    imports = loaded_dict.get_key("import", None)
+    if imports:
+        if not isinstance(imports, list):
+            raise ValueError("`import` must be a list.")
+        if base_path is None:
+            raise ValueError("Imports are not possible for non-file yaml inputs.")
+
+        for k in imports:
+            path = relative_path(base_path, k)
+            imported = read_rich_yaml(path)
+            # loaded is added to imported (i.e. it takes precedence)
+            imported.union(loaded_dict, allow_override=allow_override)
+            loaded_dict = imported
+        # 'import' key itself is no longer needed
+        loaded_dict.del_key("import")
+
+    return loaded_dict
+
+
+def to_yaml(data: AttrDict | dict, path: None | str | Path = None) -> str:
+    """Conversion to YAML.
+
+    Saves the AttrDict to the ``path`` as a YAML file or returns a YAML string
+    if ``path`` is None.
+    """
+    result = AttrDict(data).copy()
+    # Prepare YAML parsing settings
+    yaml_ = ruamel_yaml.YAML()
+    yaml_.indent = YAML_INDENT
+    yaml_.block_seq_indent = YAML_BLOCK_SEQUENCE_INDENT
+    yaml_.sort_base_mapping_type_on_output = (
```

Comment:
This is necessary for the same reason you found with `data_tables` ordering - we want the dict ordering to remain fixed on saving to file, so we cannot have it sorted (alphabetically) by `ruamel.yaml`

Reply:
I'll add a comment specifying this, then.